cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -911,63 +970,114 @@ public Boolean call() throws Exception {
 
     // switch back to alice and verify the data its fine now to verify the run record here because if the flow failed
     // to write we will not see any data
     SecurityRequestContext.setUserId(ALICE.getName());
 
-    dataSetManager = getDataset(datasetOutputSpace.toId(), "store");
-    results = dataSetManager.get();
+    DataSetManager<KeyValueTable> dataSetManager = getDataset(outputDatasetNS.getNamespaceId().toId(), "store");
+    KeyValueTable results = dataSetManager.get();
 
     for (int i = 0; i < 10; i++) {
       byte[] key = String.valueOf(i).getBytes(Charsets.UTF_8);
       Assert.assertArrayEquals(key, results.read(key));
     }
     flowManager.stop();
-    getNamespaceAdmin().delete(datasetOutputSpace.toId());
+    getNamespaceAdmin().delete(outputDatasetNS.getNamespaceId().toId());
   }
 
   @Test
-  public void testCrossNSDatasetAccessWithAuthMapReduce() throws Exception {
+  public void testCrossNSMapReduce() throws Exception {
     createAuthNamespace();
-    getNamespaceAdmin().create(new NamespaceMeta.Builder()
-                                 .setName(DatasetCrossNSAccessWithMAPApp.DATASET_INPUT_SPACE).build());
-    getNamespaceAdmin().create(new NamespaceMeta.Builder()
-                                 .setName(DatasetCrossNSAccessWithMAPApp.DATASET_OUTPUT_SPACE).build());
-    NamespaceId datasetInputSpace = new NamespaceId(DatasetCrossNSAccessWithMAPApp.DATASET_INPUT_SPACE);
-    NamespaceId datasetOutputSpace = new NamespaceId(DatasetCrossNSAccessWithMAPApp.DATASET_OUTPUT_SPACE);
-
-    addDatasetInstance(datasetInputSpace.toId(), "keyValueTable", "table1").create();
-    addDatasetInstance(datasetOutputSpace.toId(), "keyValueTable", "table2").create();
-    DataSetManager<KeyValueTable> tableManager = getDataset(datasetInputSpace.toId(), "table1");
-    KeyValueTable inputTable = tableManager.get();
-    inputTable.write("hello", "world");
-    tableManager.flush();
-
     ApplicationManager appManager = deployApplication(AUTH_NAMESPACE.toId(), DatasetCrossNSAccessWithMAPApp.class);
-    Map<String, String> argsForMR = ImmutableMap.of(DatasetCrossNSAccessWithMAPApp.INPUT_KEY, "table1",
-                                                    DatasetCrossNSAccessWithMAPApp.OUTPUT_KEY, "table2");
-    MapReduceManager mrManager = appManager.getMapReduceManager(DatasetCrossNSAccessWithMAPApp.MAPREDUCE_PROGRAM);
 
     // give BOB ALL permission in the
     grantAndAssertSuccess(AUTH_NAMESPACE, BOB, EnumSet.of(Action.ALL));
 
+    MapReduceManager mrManager = appManager.getMapReduceManager(DatasetCrossNSAccessWithMAPApp.MAPREDUCE_PROGRAM);
+
+    testCrossNSSystemDatasetAccessWithAuthMapReduce(mrManager);
+    testCrossNSDatasetAccessWithAuthMapReduce(mrManager);
+
+    appManager.stopAll();
+  }
+
+  public void testCrossNSSystemDatasetAccessWithAuthMapReduce(MapReduceManager mrManager) throws Exception {
+    addDatasetInstance(Id.Namespace.SYSTEM, "keyValueTable", "table1").create();
```

Comment:
It doesn't. Should it fail ? Isn't it that every has access to system namespace ? I am not sure about it. How does it work when you create a namespace the meta gets written in the system ns no ?